[dynamo] Lazily construct symbolic_locals

jansel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #112961
* #112962
* #111726
* #111725
* #111415
* __->__ #111717
* #111306



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

jansel committed:
[dynamo] Lazily construct symbolic_locals

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

voznesenskym committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

voznesenskym committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

voznesenskym committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

voznesenskym committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

voznesenskym committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/).

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Lazily construct symbolic_locals"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

voznesenskym commented:
NOTE! the extra graph breaks here are due to *spurious recompiles*. 

Spot checking some confirms this. For example, tts_angular recompiles because of:

`('Recompiling function forward_and_backward_pass in /data/users/voz/pytorch/benchmarks/dynamo/torchbench.py:511', "triggered by the following guard failure(s): ___check_obj_id(L['collect_outputs'], 7665344)")`

Investigating shows us this is an input bool that is unused by model code.

https://github.com/pytorch/benchmark/blob/1e03c239481d59a1485545442890295c64215f1f/userbenchmark/dynamo/dynamobench/torchbench.py#L507 

I think this PR is an unfortunate intersection of  lazy VTs and eager guards. It should be safe to land, since the subsequent PR  s fix this by not guarding on the value.

Issue to make CI clearer about recompiles (so we don't have to proxy it through the graph break count) https://github.com/pytorch/pytorch/issues/113040
